flips/20220203-capability-controllers.md

```diff
@@ -0,0 +1,390 @@
+# Capability controllers
+
+| Status        | Proposed                                           |
+| :------------ | :------------------------------------------------- |
+| **FLIP #**    | [798](https://github.com/onflow/flow/pull/798)     |
+| **Author(s)** | Janez Podhostnik (janez.podhostnik@dapperlabs.com) |
+| **Updated**   | 2022-08-22                                         |
+
+## Preface
+
+Cadence encourages a [capability-based security](d be clearer to Charlie that he received a capability t) model as described on the Flow [doc site](https://docs.onflow.org/cadence/language/capability-based-access-control). Capabilities are themselves a new concept that most Cadence programmers need to understand, but the API for syntax around Capabilities (especially the notion of “links” and “linking”), and the associated concepts of the public and private storage domains, lead to Capabilities being even more confusing and awkward to use. This proposal suggests that we could get rid of links entirely, and replace them with Capability Controllers (henceforth referred to as CapCons) which could make Capabilities easier to understand, and easier to work with.
+
+The following is a quick refresher of the current state of the Capabilities API (from the [flow doc site](https://docs.onflow.org/cadence/language/capability-based-access-control)).
+
+### Example Resource definition
+
+In the following examples let's assume that the following interface and resource type that implements the interface are defined.
+
+```cadence
+pub resource interface HasCount {
+   pub count: Int
+}
+
+pub resource Counter: HasCount {
+   pub var count: Int
+
+   pub init(count: Int) {
+       self.count = count
+   }
+}
+```
+
+The _issuer_ (`AuthAccount`) has also created an instance of this resource and saved it in its private storage.
+
+```cadence
+issuer.save(<-create Counter(count: 42), to: /storage/counter)
+```
+
+### Public Capabilities
+
+To allow anyone (read) access to the `count` field on the counter, the _issuer_ needs to create a public typed link at a chosen path that points to their stored counter resource.
```

Comment:
This is not 100% accurate. You can use `getAuthAccount` in a script and borrow this from storate and read the value without having a public capability. 

You cannot do so in transactions though.

Reply:
I think eventually we need to introduce similar functionality to transactions too